cluster/src/main/scala/io/snappydata/metrics/SnappyMetricsSystem.scala

```diff
@@ -234,8 +234,10 @@ object SnappyMetricsSystem {
   def removeTableMetricsIfTableDeleted(stringToStats: Map[String, SnappyRegionStats]): Unit = {
     for (t <- tables) {
       if (!stringToStats.exists(x => x._1 == t)) {
-        SnappyTableMetrics.removeTableMetrics(t)
-        tables -= t
+        if (t != null) {
```

Comment:
In which case, can the table-name be null? Is that a valid case or a bug?
If a valid case, it's better to check for null at the time of adding the entry itself in `tables`, in `setMetricsForTableStatDetails`?

Reply:
Added the changes you suggested